collect_app/src/androidTest/java/org/odk/collect/android/feature/instancemanagement/EditSavedFormTest.java

```diff
@@ -0,0 +1,81 @@
+package org.odk.collect.android.feature.instancemanagement;
+
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.runner.RunWith;
+import org.odk.collect.android.support.CollectTestRule;
+import org.odk.collect.android.support.TestDependencies;
+import org.odk.collect.android.support.TestRuleChain;
+import org.odk.collect.android.support.pages.MainMenuPage;
+import org.odk.collect.android.support.pages.SendFinalizedFormPage;
+
+@RunWith(AndroidJUnit4.class)
+public class EditSavedFormTest {
+
+    public static final String TEXT_FORM_NAME = "One Question";
+    public static final String TEXT_FORM_XML = "one-question.xml";
+    public static final String TEXT_QUESTION = "what is your age";
+    public static final String TEXT_ANSWER = "123";
+
+    private final CollectTestRule rule = new CollectTestRule();
+
+    final TestDependencies testDependencies = new TestDependencies();
+
+    @Rule
+    public RuleChain copyFormChain = TestRuleChain.chain(testDependencies)
+            .around(rule);
+
+    @Test
+    public void whenSubmissionSucceeds_instanceNotEditable() {
+        rule.startAtMainMenu()
+                .setServer(testDependencies.server.getURL())
+                .copyForm(TEXT_FORM_XML)
+                .startBlankForm(TEXT_FORM_NAME)
+                .answerQuestion(TEXT_QUESTION, TEXT_ANSWER)
+                .swipeToEndScreen()
+                .clickSaveAndExit()
+
+                .clickSendFinalizedForm(1)
+                .clickOnForm(TEXT_FORM_NAME)
+                .clickSendSelected()
+                .clickOK(new SendFinalizedFormPage())
+                .pressBack(new MainMenuPage())
+
+                .assertNumberOfEditableForms(0)
+                .clickEditSavedForm()
+                .assertTextDoesNotExist(TEXT_FORM_NAME)
+
+                .clickMenuFilter()
```

Comment:
This block tests the `createEditableInstancesCursorLoader` with a `charSequence`.

Reply:
A reply to myself: an alternative would be to move these assertions to a separate test (maybe in a new `EditSavedFormTest`) that checks both cases and also has a comment about it probably being a test that could probably live further down the stack.

Reply:
I went with comments for now. I initially wanted to separate them but these tests take so long that I just can't bring myself to do it for something like this! I agree it should be lower. I do think it's important to include since the initial implementation had a bug around the search behavior.